akka-http-core/src/main/scala/akka/http/impl/engine/http2/RequestParsing.scala

```diff
@@ -28,12 +29,18 @@ private[http2] object RequestParsing {
 
   @silent("use remote-address-attribute instead")
   def parseRequest(httpHeaderParser: HttpHeaderParser, serverSettings: ServerSettings, attributes: Attributes): Http2SubStream => HttpRequest = {
+
     val remoteAddressHeader: Option[`Remote-Address`] =
       if (serverSettings.remoteAddressHeader) {
         attributes.get[HttpAttributes.RemoteAddress].map(remote => model.headers.`Remote-Address`(RemoteAddress(remote.address)))
         // in order to avoid searching all the time for the attribute, we need to guard it with the setting condition
       } else None // no need to emit the remote address header
 
+    val remoteAddressAttribute: Option[RemoteAddress] =
+      if (serverSettings.remoteAddressAttribute) {
+        attributes.get[HttpAttributes.RemoteAddress].map(remote => RemoteAddress(remote.address))
```

Comment:
In retrospect perhaps we should have called the `attributes` parameter `streamAttributes` instead for contrast ;)

Reply:
Yes, very confusing when there are more kinds of attributes :/

Reply:
Oh, but that is internal so fine to change. I'll do that.